middleware/fundedAccount.js

```diff
@@ -119,15 +112,51 @@ const createFundedAccount = async (ctx) => {
     }
 };
 
+async function clearFundedAccountNeedsDeposit(ctx) {
+    const { accountId, fundedAccountNeedsDeposit } = ctx.sequelizeAccount;
+
+    if (!fundedAccountNeedsDeposit) {
+        // This is an idempotent call
+        ctx.status = 200;
+        ctx.body = { success: true };
+        return;
+    }
+
+    const nearAccount = await ctx.near.account(accountId);
+
+    const { available } = await nearAccount.getAccountBalance();
+    const availableBalanceBN = new BN(available);
+
+    if (availableBalanceBN.gt(BN_UNLOCK_FUNDED_ACCOUNT_BALANCE)) {
+        await ctx.sequelizeAccount.update({ fundedAccountNeedsDeposit: false });
+
+        ctx.status = 200;
+        ctx.body = { success: true };
+        return;
+    }
+
+    setJSONErrorResponse({
+        ctx,
+        statusCode: 200,
```

Comment:
let's not use 200 code for failed requests

Reply:
:+1: good catch -- modified this to `403 Forbidden`